src/bundler/bundle_v2.zig

```diff
@@ -5543,49 +5538,45 @@ const LinkerContext = struct {
             const fn_body = js_ast.G.FnBody{
                 .stmts = stmts.eat1(
                     js_ast.Stmt.allocate(
-                        allocator_,
+                        allocator,
                         js_ast.S.Return,
                         .{ .value = value },
                         loc,
                     ),
                 ),
                 .loc = loc,
             };
-            properties.appendAssumeCapacity(
-                .{
-                    .key = js_ast.Expr.allocate(
-                        allocator_,
-                        js_ast.E.String,
-                        .{
-                            // TODO: test emoji work as expected
-                            // relevant for WASM exports
-                            .data = alias,
-                        },
-                        loc,
-                    ),
-                    .value = js_ast.Expr.allocate(
-                        allocator_,
-                        js_ast.E.Arrow,
-                        .{ .prefer_expr = true, .body = fn_body },
-                        loc,
-                    ),
-                },
-            );
-            ns_export_symbol_uses.putAssumeCapacity(export_.data.import_ref, .{ .count_estimate = 1 });
+            properties.appendAssumeCapacity(.{
+                .key = js_ast.Expr.allocate(
+                    allocator,
+                    js_ast.E.String,
+                    .{
+                        // TODO: test emoji work as expected
+                        // relevant for WASM exports
+                        .data = alias,
+                    },
+                    loc,
+                ),
+                .value = js_ast.Expr.allocate(
+                    allocator,
+                    js_ast.E.Arrow,
+                    .{ .prefer_expr = true, .body = fn_body },
+                    loc,
+                ),
+            });
+            ns_export_symbol_uses.putAssumeCapacity(exp.data.import_ref, .{ .count_estimate = 1 });
 
             // Make sure the part that declares the export is included
-            const parts = c.topLevelSymbolsToParts(other_id, export_.data.import_ref);
+            const parts = c.topLevelSymbolsToParts(exp.data.source_index.get(), exp.data.import_ref);
             ns_export_dependencies.ensureUnusedCapacity(parts.len) catch unreachable;
-            var ptr = ns_export_dependencies.items.ptr + ns_export_dependencies.items.len;
-            ns_export_dependencies.items.len += parts.len;
 
-            for (parts, ptr[0..parts.len]) |part_id, *dependency| {
+            for (parts) |part_id| {
                 // Use a non-local dependency since this is likely from a different
                 // file if it came in through an export star
-                dependency.* = .{
-                    .source_index = export_.data.source_index,
+                ns_export_dependencies.appendAssumeCapacity(.{
```

Comment:
Do we need to use `appendAssumeCapacity` here? We cannot use the slice? Every time you call .appendAssumeCapacity, it increments the length one-by-one

Reply:
no; and im glad this was brought up since unusedCapacitySlice is even easier to read

![image](https://github.com/user-attachments/assets/9d8964da-6d10-48b2-85c7-790bc210616e)